C/services/storage/storage_api.cpp

```diff
@@ -112,25 +115,80 @@ void on_error(__attribute__((unused)) shared_ptr<HttpServer::Request> request, _
 /**
  * Wrapper function for the reading appendAPI call.
  */
-void readingAppendWrapper(shared_ptr<HttpServer::Response> response, shared_ptr<HttpServer::Request> request)
+void readingAppendWrapper(shared_ptr<HttpServer::Response> response,
+			  shared_ptr<HttpServer::Request> request)
 {
 	StorageApi *api = StorageApi::getInstance();
+#if WORKER_THREADS
+	std::atomic<int>* cnt = &(api->m_workers_count);
+	// Check rurrent number of workers and increase count
+	if (std::atomic_load(cnt) < MAX_WORKER_THREADS &&
+	    std::atomic_fetch_add(cnt, 1) < MAX_WORKER_THREADS)
```

Comment:
In this case, if MAX_WORKER_THREADS  is 5 and cnt is 4, then first part of the '&&' condition is true but second half is false and we don't allow this new thread. But we have already increased cnt to 5. Is that expected behaviour?

Reply:
@AmandeepArora I'll remove the limit and just print the counter with next commits.

This change is only for [readingQueryWrapper] removal